compiler/rustc_ast_lowering/src/delegation.rs

```diff
@@ -123,70 +124,92 @@ impl<'hir> LoweringContext<'_, 'hir> {
         })
     }
 
+    fn param_count(&self, sig_id: DefId) -> (usize, bool /*c_variadic*/) {
+        if let Some(local_sig_id) = sig_id.as_local() {
+            // Map may be filled incorrectly due to recursive delegation.
+            // Error will be emmited later in astconv.
+            match self.resolver.delegation_fn_sigs.get(&local_sig_id) {
+                Some(sig) => (sig.param_count, sig.c_variadic),
+                None => (0, false),
+            }
+        } else {
+            let sig = self.tcx.fn_sig(sig_id).skip_binder().skip_binder();
+            (sig.inputs().len(), sig.c_variadic)
```

Comment:
I've deduced that the param count as returned by `param_count` is supposed to include the `...` which definitely holds in the local case (since `DelegationFnSig.param_count` is based on the `ast::FnSig.decl.inputs.len()`) but does “`ty::FnSig.inputs()`” include the `...` (as `VaListImpl`)? I genuinely don't know. If it doesn't, we would have an off-by-one in the non-local case.

Reply:
Generally might be a good idea to add a multi-crate UI test for `fn_delegation` (`reuse`'ing items from an extern crate) or if there exists one already to extend it and test the things added in this PR.

Reply:
`ast::FnSig.decl.inputs` include `...`, and `ty::FnSig.inputs()` do not include it.

I've added a test for the cross-crate case and it's indeed failing for C variadics.